__tests__/app.test.js

```diff
@@ -407,28 +407,60 @@ describe("DELETE /api/comments/:comment_id", () => {
 
         return request(app)
-        .delete(`/api/comments/${nonExistantCommentId}`)
-        .expect(404)
-        .then(({body}) => {
-            expect(body.msg).toBe("ERROR: comment does not exist")
-        })
+            .delete(`/api/comments/${nonExistantCommentId}`)
+            .expect(404)
+            .then(({ body }) => {
+                expect(body.msg).toBe("ERROR: comment does not exist");
+            });
     });
 });
 
 describe("GET /api/users", () => {
     test("status: 200, returns an array of all user objects", () => {
         return request(app)
-        .get(`/api/users`)
-        .expect(200)
+            .get(`/api/users`)
+            .expect(200)
+            .then(({ body }) => {
+                expect(Array.isArray(body.users)).toBe(true);
+                expect(body.users).toHaveLength(4);
+                body.users.forEach((user) => {
+                    expect(typeof user.username).toBe("string");
+                    expect(typeof user.name).toBe("string");
+                    expect(typeof user.avatar_url).toBe("string");
+                });
+            });
+    });
+});
+
+describe("FEATURE (query article by topic): GET /api/articles?topic=", () => {
+    test("status: 200, returns an array of all articles with given topic", () => {
+        const topic = "mitch";
+        const numbOfArticlesWithTopic = 12;
+        return request(app)
+            .get(`/api/articles?topic=${topic}`)
+            .expect(200)
+            .then(({ body }) => {
+                expect(body.articles).toHaveLength(numbOfArticlesWithTopic);
+                body.articles.forEach((article) => {
+                    expect(typeof article.author).toBe("string");
+                    expect(typeof article.title).toBe("string");
+                    expect(typeof article.article_id).toBe("number");
+                    expect(typeof article.topic).toBe("string");
+                    expect(typeof article.created_at).toBe("string");
+                    expect(typeof article.votes).toBe("number");
+                    expect(typeof article.article_img_url).toBe("string");
+                    expect(typeof article.comment_count).toBe("number");
+                })
+            });
+    });
+    test("status: 404, reuturns an error status and msg if topic does not exist", () => {
+        const nonExiststantTopic = 'forklifts';
+        return request(app)
+        .get(`/api/articles?topic=${nonExiststantTopic}`)
+        .expect(404)
         .then(({body}) => {
-            console.log(body.users)
-            expect(Array.isArray(body.users)).toBe(true)
-            expect(body.users).toHaveLength(4);
-            body.users.forEach((user) => {
-                expect(typeof user.username).toBe('string');
-                expect(typeof user.name).toBe('string');
-                expect(typeof user.avatar_url).toBe('string')
-            })
+            expect(body.msg).toBe("ERROR: topic does not exist")
```

Comment:
Excellent. We just need one more test. A 200 where we receive an empty array for a topic which exists but has no articles.

Thanks :)